tests/test_changelog.py

```diff
@@ -509,293 +509,319 @@ def test_get_commit_tag_is_None(gitcommits, tags):
         },
-        {
-            "version": "v0.9.3",
-            "date": "2018-07-28",
-            "changes": {
-                "feat": [
-                    {
-                        "scope": "committer",
-                        "breaking": None,
-                        "message": "conventional commit is a bit more intelligent now",
-                    }
-                ]
-            },
+    },
+    {
+        "version": "v0.9.3",
+        "date": "2018-07-28",
+        "changes": {
+            "feat": [
+                {
+                    "scope": "committer",
+                    "breaking": None,
+                    "message": "conventional commit is a bit more intelligent now",
+                }
+            ]
         },
-        {
-            "version": "v0.9.2",
-            "date": "2017-11-11",
-            "changes": {
-                "refactor": [
-                    {
-                        "scope": None,
-                        "breaking": None,
-                        "message": "renamed conventional_changelog to conventional_commits, not backward compatible",
-                    }
-                ]
-            },
+    },
+    {
+        "version": "v0.9.2",
+        "date": "2017-11-11",
+        "changes": {
+            "refactor": [
+                {
+                    "scope": None,
+                    "breaking": None,
+                    "message": "renamed conventional_changelog to conventional_commits, not backward compatible",
+                }
+            ]
         },
-        {
-            "version": "v0.9.1",
-            "date": "2017-11-11",
-            "changes": {
-                "fix": [
-                    {
-                        "scope": "setup.py",
-                        "breaking": None,
-                        "message": "future is now required for every python version",
-                    }
-                ]
-            },
+    },
+    {
+        "version": "v0.9.1",
+        "date": "2017-11-11",
+        "changes": {
+            "fix": [
+                {
+                    "scope": "setup.py",
+                    "breaking": None,
+                    "message": "future is now required for every python version",
+                }
+            ]
         },
+    },
+)
+
+
+def test_generate_tree_from_commits(gitcommits, tags):
+    parser = defaults.commit_parser
+    changelog_pattern = defaults.bump_pattern
+    tree = changelog.generate_tree_from_commits(
+        gitcommits, tags, parser, changelog_pattern
     )
 
+    assert tuple(tree) == COMMITS_TREE
+
+
+@pytest.mark.parametrize(
+    "change_type_order, expected_reordering",
+    (
+        ([], {}),
+        (
+            ["BREAKING CHANGE", "refactor"],
+            {
+                2: (["refactor", "feat", "fix"], ["feat", "fix", "refactor"]),
+                3: (["BREAKING CHANGE", "refactor"], ["refactor", "BREAKING CHANGE"]),
```

Comment:
How about making `expected_reordering` a dict of dict? (e.g., `{2: {"sorted_order": ..., "original_order": ...}}`) It took me sometime to understand this test